src/build_tools/hooks_hydra.py

```diff
@@ -368,3 +371,19 @@ def pre_module_hook(self, *args, **kwargs):  # pylint: disable=unused-argument
     ############################
 
     self.cfg.enable_templating = en_templ
+
+
+def end_hook():
+    """Hook to run shortly before EasyBuild completes"""
+
+    logger = fancylogger.getLogger()
+    fancylogger.logToScreen(True, stdout=True)
+    fancylogger.setLogLevelInfo()
+
+    # submit Lmod cache job
+    if os.getenv('BUILD_TOOLS_RUN_LMOD_CACHE'):
+        partition = os.getenv('SLURM_JOB_PARTITION')
```

Comment:
Can you handle the case if it's not on in a slurm job? Skip or error out or something else?